src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8250,6 +8250,19 @@ private ErrorPropertySymbol CreateErrorPropertySymbol(ImmutableArray<PropertySym
             }
             else
             {
+                // If we're performing function pointer overload resolution and we're not in an attribute
+                // argument, then we force complete UnmanagedCallersOnly data for all methods to ensure
+                // they are loaded to be considered for calling convention comparison. If we're in an
+                // attribute argument, then it's an error case and overload resolution will treat it as
+                // matching to suppress potentially spurious diagnostics
+                if (!InAttributeArgument && isFunctionPointerResolution)
+                {
+                    foreach (var method in methodGroup.Methods)
+                    {
+                        method.ForceCompleteUnmanagedCallersOnlyAttribute();
```

Comment:
>method.ForceCompleteUnmanagedCallersOnlyAttribute() [](start = 24, length = 51)

Feels wrong to call this for a PE symbol.

Reply:
I'm going to refactor the existing `UnmanagedCallersOnlyAttributeData` property to be a method with a `forceComplete` parameter, and do a follow up PR to make PEMethodSymbol not do a full attribute load in the implementation of that method.

---
In reply to: [489010581](https://github.com/dotnet/roslyn/pull/47651#discussion_r489010581) [](ancestors = 489010581)